subsys/bluetooth/host/l2cap_internal.h

```diff
@@ -381,3 +381,16 @@ void bt_l2cap_register_ecred_cb(const struct bt_l2cap_ecred_cb *cb);
 
 /* Returns a server if it exists for given psm. */
 struct bt_l2cap_server *bt_l2cap_server_lookup_psm(uint16_t psm);
+
+/** Override tx mtu (ie. max SDU payload size) for the given channel.
+ *
+ *  This is used to make the L2CAP layer enforce a max message size on behalf of
+ *  an upper layer, as a convenience.
+ *
+ *  Should work in the @ref bt_l2cap_chan_ops.connected @ref
+ *  bt_l2cap_chan_ops.reconfigured callbacks.
+ */
+static inline void bt_l2cap_set_tx_mtu(struct bt_l2cap_le_chan *ch, uint16_t mtu)
```

Comment:
Should this function be used all the places where we do `tx.mtu = mtu`?

Reply:
Yes, when outside of `l2cap.c` at least. But I wanted to keep this PR minimal. I can post a follow up later.

I'll take this chance to explain why it's nice to define a function like this, even though the body is obvious: It makes it clear that this "hack" is OK by the L2CAP layer. I think `chan.tx.mtu` was originally intended to be what I name `chan.remote.rx.mtu`. In that case, directly setting `chan.tx.mtu` looks like writing to a "read-only" value. We are making it clear that `chan.tx.mtu` is something different than `chan.remote.rx.mtu`.